Panic on boot after poweroff

Repro: start RancherOS, log in, `sudo poweroff`, boot the same machine again, panic!
Reboots can also trigger the same panic, but I can't confirm it happens all the time. 
Affects the current master.

```
INFO[0000] [1/15] [system-volumes]: Started             
INFO[0001] [2/15] [user-volumes]: Started               
INFO[0001] [3/15] [docker-volumes]: Started             
INFO[0001] [4/15] [command-volumes]: Started            
INFO[0001] [5/15] [syslog]: Started                     
INFO[0001] [6/15] [all-volumes]: Started                
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x140 pc=0x5c00ce]

goroutine 21 [running]:
github.com/rancherio/os/docker.(*Container).start(0xc20802e300, 0x100, 0x0)
        /source/gopath/src/github.com/rancherio/os/docker/container.go:575 +0x65e
github.com/rancherio/os/docker.(*Container).StartAndWait(0xc20802e300, 0xa53a30)
        /source/gopath/src/github.com/rancherio/os/docker/container.go:347 +0x32
github.com/rancherio/os/docker.(*containerBasedService).Up(0xc208095050, 0x0, 0x0)
        /source/gopath/src/github.com/rancherio/os/docker/factory.go:50 +0x42f
github.com/rancherio/rancher-compose/librcompose/project.(*serviceWrapper).Start(0xc20802e2a0, 0xc208094570)
        /source/Godeps/_workspace/src/github.com/rancherio/rancher-compose/librcompose/project/service-wrapper.go:72 +0x5f9
created by github.com/rancherio/rancher-compose/librcompose/project.(*Project).startAll
        /source/Godeps/_workspace/src/github.com/rancherio/rancher-compose/librcompose/project/project.go:122 +0x1a5

goroutine 1 [runnable]:
sync.(*WaitGroup).Wait(0xc20802e2c0)
        /usr/local/go/src/sync/waitgroup.go:132 +0x169
github.com/rancherio/rancher-compose/librcompose/project.(*serviceWrapper).Wait(0xc20802e2a0, 0x0, 0x0)
        /source/Godeps/_workspace/src/github.com/rancherio/rancher-compose/librcompose/project/service-wrapper.go:84 +0x46
github.com/rancherio/rancher-compose/librcompose/project.(*Project).startAll(0xc208080580, 0xc208094570, 0x0, 0x0)
        /source/Godeps/_workspace/src/github.com/rancherio/rancher-compose/librcompose/project/project.go:128 +0x250
github.com/rancherio/rancher-compose/librcompose/project.(*Project).Up(0xc208080580, 0x0, 0x0)
        /source/Godeps/_workspace/src/github.com/rancherio/rancher-compose/librcompose/project/project.go:101 +0xa5
github.com/rancherio/os/docker.RunServices(0xa40210, 0xb, 0xc208020380, 0xc2080942a0, 0x0, 0x0)
        /source/gopath/src/github.com/rancherio/os/docker/services.go:131 +0xa10
github.com/rancherio/os/init.runContainers(0xc208020380, 0x0, 0x0)
        /source/gopath/src/github.com/rancherio/os/init/sysinit.go:110 +0x5c
github.com/rancherio/os/config.RunInitFuncs(0xc208020380, 0xc2080c7e58, 0x5, 0x5, 0x0, 0x0)
        /source/gopath/src/github.com/rancherio/os/config/init.go:14 +0x1f4
github.com/rancherio/os/init.SysInit(0x0, 0x0)
        /source/gopath/src/github.com/rancherio/os/init/sysinit.go:165 +0x16b
github.com/rancherio/os/cmd/sysinit.Main()
        /source/gopath/src/github.com/rancherio/os/cmd/sysinit/sysinit.go:9 +0x1f
github.com/docker/docker/pkg/reexec.Init(0xa78ff0)
        /source/Godeps/_workspace/src/github.com/docker/docker/pkg/reexec/reexec.go:26 +0x71
main.main()
        /source/main.go:53 +0x20d

goroutine 5 [syscall]:
os/signal.loop()
        /usr/local/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
        /usr/local/go/src/os/signal/signal_unix.go:27 +0x35

goroutine 7 [chan receive]:
github.com/rancherio/os/docker.func·002()
        /source/gopath/src/github.com/rancherio/os/docker/services.go:119 +0x6d
created by github.com/rancherio/os/docker.RunServices
        /source/gopath/src/github.com/rancherio/os/docker/services.go:124 +0x83f

goroutine 8 [semacquire]:
sync.(*WaitGroup).Wait(0xc20802ec80)
        /usr/local/go/src/sync/waitgroup.go:132 +0x169
github.com/rancherio/rancher-compose/librcompose/project.(*serviceWrapper).Wait(0xc20802ec60, 0x0, 0x0)
        /source/Godeps/_workspace/src/github.com/rancherio/rancher-compose/librcompose/project/service-wrapper.go:84 +0x46
github.com/rancherio/rancher-compose/librcompose/project.(*serviceWrapper).Start(0xc20802e540, 0xc208094570)
        /source/Godeps/_workspace/src/github.com/rancherio/rancher-compose/librcompose/project/service-wrapper.go:58 +0x3b3
created by github.com/rancherio/rancher-compose/librcompose/project.(*Project).startAll
        /source/Godeps/_workspace/src/github.com/rancherio/rancher-compose/librcompose/project/project.go:122 +0x1a5

goroutine 11 [IO wait]:
net.(*pollDesc).Wait(0xc208101b10, 0x72, 0x0, 0x0)
        /usr/local/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc208101b10, 0x0, 0x0)
        /usr/local/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc208101ab0, 0xc2080f7000, 0x1000, 0x1000, 0x0, 0x7f087b1f6e18, 0xc20807aad8)
        /usr/local/go/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc20802a208, 0xc2080f7000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /usr/local/go/src/net/net.go:121 +0xdc
bufio.(*Reader).fill(0xc20802f440)
        /usr/local/go/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).ReadSlice(0xc20802f440, 0x4f0d0a, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/local/go/src/bufio/bufio.go:295 +0x257
bufio.(*Reader).ReadLine(0xc20802f440, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/local/go/src/bufio/bufio.go:324 +0x62
net/textproto.(*Reader).readLineSlice(0xc2080e4f00, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/local/go/src/net/textproto/reader.go:55 +0x9e
net/textproto.(*Reader).ReadLine(0xc2080e4f00, 0x0, 0x0, 0x0, 0x0)
        /usr/local/go/src/net/textproto/reader.go:36 +0x4f
net/http.ReadResponse(0xc20802f440, 0xc20808a680, 0xc208075788, 0x0, 0x0)
        /usr/local/go/src/net/http/response.go:123 +0x10c
net/http/httputil.(*ClientConn).Read(0xc208080f20, 0xc20808a680, 0x0, 0x0, 0x0)
        /usr/local/go/src/net/http/httputil/persist.go:404 +0x47b
net/http/httputil.(*ClientConn).Do(0xc208080f20, 0xc20808a680, 0x0, 0x0, 0x0)
        /usr/local/go/src/net/http/httputil/persist.go:428 +0x84
github.com/fsouza/go-dockerclient.(*Client).do(0xc20810b280, 0xa0dc70, 0x3, 0xa0aa70, 0x5, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /source/Godeps/_workspace/src/github.com/fsouza/go-dockerclient/client.go:317 +0x435
github.com/fsouza/go-dockerclient.(*Client).Info(0xc20810b280, 0x22, 0x0, 0x0)
        /source/Godeps/_workspace/src/github.com/fsouza/go-dockerclient/misc.go:31 +0x8d
github.com/rancherio/os/docker.NewClient(0xab5ab0, 0x22, 0xa4f470, 0x0, 0x0)
        /source/gopath/src/github.com/rancherio/os/docker/client.go:33 +0xc7
github.com/rancherio/os/docker.(*Container).Delete(0xc20802e720, 0x15)
        /source/gopath/src/github.com/rancherio/os/docker/container.go:395 +0xbf
github.com/rancherio/os/docker.(*Container).start(0xc20802e720, 0x100, 0x0)
        /source/gopath/src/github.com/rancherio/os/docker/container.go:543 +0x3bf
github.com/rancherio/os/docker.(*Container).StartAndWait(0xc20802e720, 0xa53a30)
        /source/gopath/src/github.com/rancherio/os/docker/container.go:347 +0x32
github.com/rancherio/os/docker.(*containerBasedService).Up(0xc208094510, 0x0, 0x0)
        /source/gopath/src/github.com/rancherio/os/docker/factory.go:50 +0x42f
github.com/rancherio/rancher-compose/librcompose/project.(*serviceWrapper).Start(0xc20802e6c0, 0xc208094570)
        /source/Godeps/_workspace/src/github.com/rancherio/rancher-compose/librcompose/project/service-wrapper.go:72 +0x5f9
created by github.com/rancherio/rancher-compose/librcompose/project.(*Project).startAll
        /source/Godeps/_workspace/src/github.com/rancherio/rancher-compose/librcompose/project/project.go:122 +0x1a5

goroutine 13 [semacquire]:
sync.(*WaitGroup).Wait(0xc20802ef20)
        /usr/local/go/src/sync/waitgroup.go:132 +0x169
github.com/rancherio/rancher-compose/librcompose/project.(*serviceWrapper).Wait(0xc20802ef00, 0x0, 0x0)
        /source/Godeps/_workspace/src/github.com/rancherio/rancher-compose/librcompose/project/service-wrapper.go:84 +0x46
github.com/rancherio/rancher-compose/librcompose/project.(*serviceWrapper).Start(0xc20802e780, 0xc208094570)
        /source/Godeps/_workspace/src/github.com/rancherio/rancher-compose/librcompose/project/service-wrapper.go:58 +0x3b3
created by github.com/rancherio/rancher-compose/librcompose/project.(*Project).startAll
        /source/Godeps/_workspace/src/github.com/rancherio/rancher-compose/librcompose/project/project.go:122 +0x1a5

goroutine 16 [runnable]:
sync.(*WaitGroup).Wait(0xc20802e2c0)
        /usr/local/go/src/sync/waitgroup.go:132 +0x169
github.com/rancherio/rancher-compose/librcompose/project.(*serviceWrapper).Wait(0xc20802e2a0, 0x0, 0x0)
        /source/Godeps/_workspace/src/github.com/rancherio/rancher-compose/librcompose/project/service-wrapper.go:84 +0x46
github.com/rancherio/rancher-compose/librcompose/project.(*serviceWrapper).Start(0xc20802eae0, 0xc208094570)
        /source/Godeps/_workspace/src/github.com/ran[    5.623038] serial8250: too much work for irq4
cherio/rancher-compose/librcompose/project/service-wrapper.go:58 +0x3b3
created by github.com/rancherio/rancher-compose/librcompose/project.(*Project).startAll
        /source/Godeps/_workspace/src/github.com/rancherio/rancher-compose/librcompose/project/project.go:122 +0x1a5

goroutine 17 [semacquire]:
sync.(*WaitGroup).Wait(0xc20802ef20)
        /usr/local/go/src/sync/waitgroup.go:132 +0x169
github.com/rancherio/rancher-compose/librcompose/project.(*serviceWrapper).Wait(0xc20802ef00, 0x0, 0x0)
        /source/Godeps/_workspace/src/github.com/rancherio/rancher-compose/librcompose/project/service-wrapper.go:84 +0x46
github.com/rancherio/rancher-compose/librcompose/project.(*serviceWrapper).Start(0xc20802ec00, 0xc208094570)
        /source/Godeps/_workspace/src/github.com/rancherio/rancher-compose/librcompose/project/service-wrapper.go:58 +0x3b3
created by github.com/rancherio/rancher-compose/librcompose/project.(*Project).startAll
        /source/Godeps/_workspace/src/github.com/rancherio/rancher-compose/librcompose/project/project.go:122 +0x1a5

goroutine 18 [runnable]:
sync.(*WaitGroup).Wait(0xc20802e2c0)
        /usr/local/go/src/sync/waitgroup.go:132 +0x169
github.com/rancherio/rancher-compose/librcompose/project.(*serviceWrapper).Wait(0xc20802e2a0, 0x0, 0x0)
        /source/Godeps/_workspace/src/github.com/rancherio/rancher-compose/librcompose/project/service-wrapper.go:84 +0x46
github.com/rancherio/rancher-compose/librcompose/project.(*serviceWrapper).Start(0xc20802ef00, 0xc208094570)
        /source/Godeps/_workspace/src/github.com/rancherio/rancher-compose/librcompose/project/service-wrapper.go:58 +0x3b3
created by github.com/rancherio/rancher-compose/librcompose/project.(*Project).startAll
        /source/Godeps/_workspace/src/github.com/rancherio/rancher-compose/librcompose/project/project.go:122 +0x1a5

goroutine 19 [semacquire]:
sync.(*WaitGroup).Wait(0xc20802eb00)
        /usr/local/go/src/sync/waitgroup.go:132 +0x169
github.com/rancherio/rancher-compose/librcompose/project.(*serviceWrapper).Wait(0xc20802eae0, 0x0, 0x0)
        /source/Godeps/_workspace/src/github.com/rancherio/rancher-compose/librcompose/project/service-wrapper.go:84 +0x46
github.com/rancherio/rancher-compose/librcompose/project.(*serviceWrapper).Start(0xc20802ec60, 0xc208094570)
        /source/Godeps/_workspace/src/github.com/rancherio/rancher-compose/librcompose/project/service-wrapper.go:58 +0x3b3
created by github.com/rancherio/rancher-compose/librcompose/project.(*Project).startAll
        /source/Godeps/_workspace/src/github.com/rancherio/rancher-compose/librcompose/project/project.go:122 +0x1a5

goroutine 22 [IO wait]:
net.(*pollDesc).Wait(0xc208101fe0, 0x72, 0x0, 0x0)
        /usr/local/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc208101fe0, 0x0, 0x0)
        /usr/local/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc208101f80, 0xc20810f000, 0x1000, 0x1000, 0x0, 0x7f087b1f6e18, 0xc20807b608)
        /usr/local/go/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc20802a298, 0xc20810f000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /usr/local/go/src/net/net.go:121 +0xdc
bufio.(*Reader).fill(0xc20802f740)
        /usr/local/go/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).ReadSlice(0xc20802f740, 0x4f0d0a, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/local/go/src/bufio/bufio.go:295 +0x257
bufio.(*Reader).ReadLine(0xc20802f740, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/local/go/src/bufio/bufio.go:324 +0x62
net/textproto.(*Reader).readLineSlice(0xc2080e5770, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/local/go/src/net/textproto/reader.go:55 +0x9e
net/textproto.(*Reader).ReadLine(0xc2080e5770, 0x0, 0x0, 0x0, 0x0)
        /usr/local/go/src/net/textproto/reader.go:36 +0x4f
net/http.ReadResponse(0xc20802f740, 0xc20808a820, 0xc2080c9838, 0x0, 0x0)
        /usr/local/go/src/net/http/response.go:123 +0x10c
net/http/httputil.(*ClientConn).Read(0xc208081290, 0xc20808a820, 0x0, 0x0, 0x0)
        /usr/local/go/src/net/http/httputil/persist.go:404 +0x47b
net/http/httputil.(*ClientConn).Do(0xc208081290, 0xc20808a820, 0x0, 0x0, 0x0)
        /usr/local/go/src/net/http/httputil/persist.go:428 +0x84
github.com/fsouza/go-dockerclient.(*Client).do(0xc20810a880, 0xa0ca90, 0x6, 0xc20802f680, 0x54, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /source/Godeps/_workspace/src/github.com/fsouza/go-dockerclient/client.go:317 +0x435
github.com/fsouza/go-dockerclient.(*Client).RemoveContainer(0xc20810a880, 0xc2080e9080, 0x40, 0x100, 0x0, 0x0)
        /source/Godeps/_workspace/src/github.com/fsouza/go-dockerclient/container.go:587 +0x154
github.com/rancherio/os/docker.(*Container).Delete(0xc20802e8a0, 0x15)
        /source/gopath/src/github.com/rancherio/os/docker/container.go:403 +0x191
github.com/rancherio/os/docker.(*Container).start(0xc20802e8a0, 0x100, 0x0)
        /source/gopath/src/github.com/rancherio/os/docker/container.go:543 +0x3bf
github.com/rancherio/os/docker.(*Container).StartAndWait(0xc20802e8a0, 0xa53a30)
        /source/gopath/src/github.com/rancherio/os/docker/container.go:347 +0x32
github.com/rancherio/os/docker.(*containerBasedService).Up(0xc2080958c0, 0x0, 0x0)
        /source/gopath/src/github.com/rancherio/os/docker/factory.go:50 +0x42f
github.com/rancherio/rancher-compose/librcompose/project.(*serviceWrapper).Start(0xc20802e840, 0xc208094570)
        /source/Godeps/_workspace/src/github.com/rancherio/rancher-compose/librcompose/project/service-wrapper.go:72 +0x5f9
created by github.com/rancherio/rancher-compose/librcompose/project.(*Project).startAll
        /source/Godeps/_workspace/src/github.com/rancherio/rancher-compose/librcompose/project/project.go:122 +0x1a5
[  107.768302] random: nonblocking pool is initialized
```

--- a/docker/container.go
+++ b/docker/container.go
@@ -532,7 +532,6 @@ func (c *Container) start(createOnly, wait bool) *Container {
 		return c.returnErr(err)
 	}
 
-	container := c.Container
 	created := false
 
 	opts, err := c.getCreateOpts(client)
@@ -566,14 +565,12 @@ func (c *Container) start(createOnly, wait bool) *Container {
 			return c.returnErr(err)
 		}
 
-		container, err = client.CreateContainer(*opts)
+		c.Container, err = client.CreateContainer(*opts)
 		created = true
 		if err != nil {
 			return c.returnErr(err)
 		}
 	}
-
-	c.Container = container
 
 	hostConfig := c.Container.HostConfig
 	if created {

